lib/core/controlplane.rb

```diff
@@ -383,16 +373,30 @@ def parse_apply_result(result) # rubocop:disable Metrics/CyclomaticComplexity, M
 
   private
 
-  def perform(cmd)
+  def build_command(cmd, show_stdout: true, show_stderr: true, show_in_tests: false)
+    show_stdout &&= !Shell.should_hide_output?
+
+    show_for_env = ENV["RAILS_ENV"] != "test" || show_in_tests
+    cmd += " > /dev/null" unless show_stdout && show_for_env
+    cmd += " 2> /dev/null" unless show_stderr && show_for_env
+
+    cmd
+  end
+
+  def perform(cmd, show_stdout: true, show_stderr: true, show_in_tests: false)
+    cmd = build_command(cmd, show_stdout: show_stdout, show_stderr: show_stderr, show_in_tests: show_in_tests)
+
     Shell.debug("CMD", cmd)
 
-    system(cmd)
+    Kernel.system(cmd)
   end
 
-  def perform!(cmd, sensitive_data_pattern: nil)
+  def perform!(cmd, show_stdout: true, show_stderr: true, show_in_tests: false, sensitive_data_pattern: nil)
+    cmd = build_command(cmd, show_stdout: show_stdout, show_stderr: show_stderr, show_in_tests: show_in_tests)
+
     Shell.debug("CMD", cmd, sensitive_data_pattern: sensitive_data_pattern)
 
-    system(cmd) || exit(1)
+    Kernel.system(cmd) || exit(1)
```

Comment:
Better debugging to explain what's going on, at least that the program is exiting because of non-zero exit code from cmd